File: galgamecommon/src/main/scala/com/goexp/galgame/common/model/game/guide/GameGuide.java
package com.goexp.galgame.common.model.game.guide;

import java.util.Objects;
import java.util.StringJoiner;

public class GameGuide {
    public String id;
    public String title;
    public String href;
    public String html;
    public DataFrom from;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof GameGuide)) return false;
        GameGuide gameGuide = (GameGuide) o;
        return id.equals(gameGuide.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return new StringJoiner(", ", GameGuide.class.getSimpleName() + "[", "]")
                .add("id='" + id + "'")
                .add("title='" + title + "'")
                .add("href='" + href + "'")
                .add("from=" + from)
                .toString();
    }

}

File: common/src/main/scala/com/goexp/galgame/common/model/game/guide/GameGuide.scala
package com.goexp.galgame.common.model.game.guide

import scala.beans.BeanProperty

class GameGuide {
  var id: String = _
  @BeanProperty var title: String = _
  @BeanProperty var href: String = _
  var html: String = _
  @BeanProperty var from: DataFrom = _


  def canEqual(other: Any): Boolean = other.isInstanceOf[GameGuide]

  override def equals(other: Any): Boolean = other match {
    case that: GameGuide =>
      (that canEqual this) &&
        id == that.id
    case _ => false
  }

  override def hashCode(): Int = {
    val state = Seq(id)
    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
  }
}